[#103313294] add postgres service to aws

mtekel opened the pull request:
https://www.pivotaltracker.com/n/projects/1275640/stories/103313294

Add postgreSQL shared VM service into CF:
- add dedicated 'postgres' server to host the shared plan
- build the broker: we build on the bastion host, it requires maven. CF java buildpack only supports deploying binaries...
- deploy and register PSQL broker: use cf cli to push the app and register the broker and service
# Testing

`make aws DEPLOY_ENV=<your environment>`

At the end of deploy you should be able to list PSQL service in marketplace: `cf m`:

```
cf m
Getting services from marketplace in org admin / space admin as admin...
OK

service      plans                    description   
PostgreSQL   Basic PostgreSQL Plan*   PostgreSQL on shared instance.   

* These service plans have an associated cost. Creating a service instance will incur this cost.
```

You can also check service information with `cf m -s PostgreSQL`:

```
ubuntu@michael-cf-bastion:~$ cf m -s PostgreSQL
Getting service plan information for service PostgreSQL as admin...
OK

service plan            description                                                                        free or paid   
Basic PostgreSQL Plan   A PG plan providing a single database on a shared instance with limited storage.   paid   
```

You can create service instance and bind it to some app and then check environment variables, you should get st. like

```
cf env flaskapp
Getting env variables for app flaskapp in org admin / space admin as admin...
OK

System-Provided:
{
 "VCAP_SERVICES": {
  "PostgreSQL": [
   {
    "credentials": {
     "uri": "postgres://cff7468f-9156-4044-8acc-6a7e9f777108:2gjb02o6katmfrdl0p3lmvc7a1@10.0.0.253:5432/cff7468f-9156-4044-8acc-6a7e9f777108"
    },
    "label": "PostgreSQL",
    "name": "mypostgres",
    "plan": "Basic PostgreSQL Plan",
    "tags": [
     "PostgreSQL",
     "Database storage"
    ]
   }
  ]
 }
}
```

The flask app has been [patched](https://github.com/alphagov/flask-sqlalchemy-postgres-heroku-example/commit/716513f7ecc765337685a1529685a93bba97b4d9) to support the CF VCAP services natively rather than requiring them to be manually specified as in https://www.pivotaltracker.com/n/projects/1275640/stories/102980618

You can also check service instance information by `cf service <servicename>`:

```
ubuntu@michael-cf-bastion:~$ cf service mypostgres

Service instance: mypostgres
Service: PostgreSQL
Plan: Basic PostgreSQL Plan
Description: PostgreSQL on shared instance.
Documentation url: http://mendix.com/postgresql
Dashboard: 

Last Operation
Status: create succeeded
Message: 
Started: 2015-09-16T10:14:41Z
Updated: 2015-09-16T10:14:41Z
```
# Testing

Not @mtekel or @Jonty

mtekel committed:
psql deploy script: parametrize CF credentials

CF passwords in GCE and AWS are different, parametrize. Also fix a few typos and whitespace formatting.

mtekel committed:
aws provision: cleanup

Do a bit of cleanup of aws provision: formatting, remove irrelevant comments and do few steps to unifiy it more with GCE provision.

mtekel committed:
psql deploy: log in and create org

On AWS you have to log in in a special way, because there's no admin org. Also, create that org.

mtekel committed:
deploy psql: configure security groups

GCE comes with security groups that open internal network communication by default. AWS does not, create a CF security group that explicityl allows port 5432 on local networks and apply that group to admin space in the admin org so that we can deploy psql broker.

mtekel committed:
AWS: add postgres server manifest

Add a manifest that deploys a single server for postgres. The name 'postgres' is unified with GCE.

keymon commented:
@mtekel now there is a scripts directory, do you mind move the script `deploy_psql_broker.sh` there?

mtekel committed:
Source psql install script from scrips dir

Use same directory and same script for both AWS and GCE. The script is now located in the scripts directory.

keymon commented:
My only critic is that we copy the file with a provider in the bastion, but that is not a big deal. We can fix that in a different story.

LGTM, and works fine, merging :+1: :)
